packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.stories.tsx

```diff
@@ -6,39 +6,101 @@ import React from 'react';
 import AccountRecoveryResetPassword, {
   AccountRecoveryResetPasswordProps,
 } from '.';
-import AppLayout from '../../../components/AppLayout';
 import { LocationProvider } from '@reach/router';
 import { Meta } from '@storybook/react';
-import { MOCK_ACCOUNT } from '../../../models/mocks';
 import { withLocalization } from '../../../../.storybook/decorators';
-import { LinkStatus } from '../../../lib/types';
+import { AppContext, AppContextValue } from '../../../models';
+import {
+  mockUrlSearchContext,
+  mockUrlHashContext,
+  mockStorageContext,
+  mockAccount,
+  mockRelierFactory,
+  mockLocationContext,
+} from './mocks';
+import { mockAppContext } from '../../../models/mocks';
+import { AuthUiErrors } from '../../../lib/auth-errors/auth-errors';
 
 export default {
   title: 'Pages/ResetPassword/AccountRecoveryResetPassword',
   component: AccountRecoveryResetPassword,
   decorators: [withLocalization],
 } as Meta;
 
-const storyWithProps = (props?: Partial<AccountRecoveryResetPasswordProps>) => {
-  const story = () => (
-    <LocationProvider>
-      <AppLayout>
-        <AccountRecoveryResetPassword
-          email={MOCK_ACCOUNT.primaryEmail.email}
-          linkStatus="valid"
-        />
-      </AppLayout>
-    </LocationProvider>
+const storyWithProps = (
+  ctx: AppContextValue,
+  props: AccountRecoveryResetPasswordProps
+) => {
+  return (
+    <AppContext.Provider value={ctx}>
+      <LocationProvider>
+        <AccountRecoveryResetPassword {...props} />
+      </LocationProvider>
+    </AppContext.Provider>
   );
-  return story;
 };
 
-export const WithValidLink = storyWithProps();
+function setup() {
+  const account = mockAccount();
+  const navigate = async (to: string | number, opts?: {}) => {
+    console.log('Would navigate to', to, opts);
+  };
+  const urlSearchContext = mockUrlSearchContext();
+  const urlHashContext = mockUrlHashContext();
+  const storageContext = mockStorageContext();
+  const locationContext = mockLocationContext();
+  const relierFactory = mockRelierFactory(
+    urlSearchContext,
+    urlHashContext,
+    storageContext
+  );
+
+  const ctx = mockAppContext({
+    urlSearchContext,
+    urlHashContext,
+    storageContext,
+    account,
+    relierFactory,
+  });
 
-export const WithBrokenLink = storyWithProps({
-  linkStatus: LinkStatus.damaged,
-});
+  return {
+    ctx,
+    props: {
+      overrides: {
+        navigate,
+        urlSearchContext,
+        locationContext,
+      },
+    },
+    account,
+    urlSearchContext,
+    locationContext,
+  };
+}
 
-export const WithExpiredLink = storyWithProps({
-  linkStatus: LinkStatus.expired,
-});
+export const WithValidLink = () => {
+  const { props, ctx } = setup();
+  return storyWithProps(ctx, props);
+};
+
+export const WithExpiredLink = () => {
```

Comment:
I checkout out this branch and noticed that this story is loading the valid link view. At the moment, not sure if this is just storybook or if the link validity check might not be working.
![image](https://user-images.githubusercontent.com/22231637/224384391-98923af4-f0a8-41da-a79e-b11f2a6e0b74.png)

Reply:
If you wait too long before clicking on the email link, wouldn't it go directly to the link expired view too? I had understood that it would check twice - on render and on submit.

Reply:
I'm basing this assumption on this check in CompleteResetPassword: 
https://github.com/mozilla/fxa/blob/6a1e2ee612f57385bb8d4422f1fc5c91696b1467/packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.tsx#L114-L128

Reply:
The preceding page, AccountRecoveryConfirmKey does a link check on load and on submit. This would catch the scenario where a user waits too long to click the email link, and also handle the scenario where the user leaves their page open for so long the link expires. (AFAIK no email links to the AccountRecoveryResetPassword page directly.) 

We could check again, just to keep it symmetric, but it seemed like a redundant call that just determines a state that would have been checked milliseconds prior.